vslib/src/sai_vs_generic_create.cpp

```diff
@@ -0,0 +1,213 @@
+#include "sai_vs.h"
+
+ObjectHash g_objectHash;
+
+uint64_t g_real_id = 0;
+
+sai_object_id_t vs_create_real_object_id(
+        _In_ sai_object_type_t object_type)
+{
+    SWSS_LOG_ENTER();
+
+    g_real_id++;
+
+    sai_object_id_t objectId = (((sai_object_id_t)object_type) << 48) | g_real_id;
+
+    SWSS_LOG_DEBUG("created RID %llx", objectId);
+
+    return objectId;
+}
+
+/**
+ * Routine Description:
+ *     @brief  Query sai object type.
+ *
+ * Arguments:
+ *     @param[in] sai_object_id
+ *
+ * Return Values:
+ *    @return  Return SAI_OBJECT_TYPE_NULL when sai_object_id is not valid.
+ *             Otherwise, return a valid sai object type SAI_OBJECT_TYPE_XXX
+ */
+sai_object_type_t sai_object_type_query(_In_ sai_object_id_t sai_object_id)
+{
+    return (sai_object_type_t)(sai_object_id >> 48);
+}
+
+/**
+ *   Routine Description:
+ *    @brief Generic create method
+ *
+ *  Arguments:
+ *  @param[out] acl_table_id - the the acl table id
+ *  @param[in] attr_count - number of attributes
+ *  @param[in] attr_list - array of attributes
+ *
+ *  Return Values:
+ *    @return  SAI_STATUS_SUCCESS on success
+ *             Failure status code on error
+ */
+sai_status_t internal_vs_generic_create(
+        _In_ sai_object_type_t object_type,
+        _In_ const std::string &serialized_object_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list)
+{
+    std::lock_guard<std::mutex> lock(g_mutex);
+
+    SWSS_LOG_ENTER();
+
+    auto it = g_objectHash.find(serialized_object_id);
```

Comment:
It out param, take a look at line 108 and 117 and 122, 
also look at line 64, this is just safety check